src/gpu/optixPrograms.cu

```diff
@@ -53,17 +52,11 @@ Vec3f decodePayloadVec3f(const Vec3fPayload& src)
 	};
 }
 
-
 template<bool isFinite>
 __forceinline__ __device__
-void saveRayResult(Vec3f* xyz=nullptr)
+void saveRayResult(const Vec3f* xyz=nullptr, const Vec3f* origin=nullptr)
 {
 	const int rayIdx = optixGetLaunchIndex().x;
-	Vec3f origin = decodePayloadVec3f({
-		optixGetPayload_0(),
-		optixGetPayload_1(),
-		optixGetPayload_2()
-	});
```

Comment:
`optixGetPayload_0` is not available when called from `__raygen__` function, hence it was moved to the callee side.